### PR TITLE
Properly handle controller parameter.

### LINKED ISSUE #9: Support controller as array in `instances` API calls
Code should support controller as a string or an array. Currently the code does a strange mix of these two things (queries using an array but the input param is treated only as a string).

### PR DESCRIPTION
**Pull request by mattcollier**
Fixes #9.

### CODE REVIEW

File: lib/instances.js [hunk 1]
```diff
@@ -157,10 +158,15 @@ exports.getControllerKey = async ({id}) => {
 
 // set issuer and capabilities (zcaps) for an instance
 exports.setIssuer = async ({id, controller, issuer, zcaps}) => {
```
Comment (gannan08) [inline lib/instances.js]: This should be addressed in a different PR.

```suggestion
exports.setIssuerInstance = async ({id, controller, issuer, zcaps}) => {
```

File: lib/instances.js [hunk 2]
```diff
@@ -157,10 +158,15 @@ exports.getControllerKey = async ({id}) => {
...
+  assert.string(id, 'id');
   const query = {'instance.id': id};
   if(controller) {
+    if(!Array.isArray(controller)) {
```
Comment (gannan08) [inline lib/instances.js]: Was this done to handle the case for when `controller` is an Array? I don't think the API should support multiple controllers at once trying to set properties on the Issuer Instance. Why not assert the controller property to be a string.

File: lib/instances.js (hunk 1 above)
Comment (mattcollier) [reply lib/instances.js]: We're in the `instances` API here, isn't the `Instance` you tacked on the end implicit?

File: lib/instances.js (hunk 2 above)
Comment (mattcollier) [reply lib/instances.js]: @dlongley this one is for you.

File: lib/instances.js (hunk 1 above)
Comment (dlongley) [reply lib/instances.js]: I agree with @mattcollier

Comment (gannan08) [reply lib/instances.js]: I did not realize the file was named appropriately. This suggestion should be ignored.

File: lib/instances.js (hunk 2 above)
Comment (mattcollier) [reply lib/instances.js]: if controller is an array in the database, you don't use `$in`, doing `controller: controller` just works.  Is that what is wanted here?

Comment (gannan08) [reply lib/instances.js]: @mattcollier That should work.

https://docs.mongodb.com/manual/tutorial/query-arrays/#query-an-array-for-an-element

Comment (mattcollier) [reply lib/instances.js]: New implementation now.
